src/api/resources/openApi.json

```diff
@@ -369,19 +369,26 @@
                 "type": "object",
                 "additionalProperties": true,
                 "properties": {
-                  "points": {
+                  "coordinatesMeta": {
                     "type": "array",
                     "title": "Route point details",
                     "description": "Metadata for each point within the route. Array length MUST be the same as length of coordinates array!",
                     "items": {
-                      "$ref": "#/components/schemas/RoutePointMeta"
+                        "oneOf": [
```

Comment:
sorry for my late reply: can't we have both: route points name inline and Href to the real waypoint? that way a UI gets all the data in one go and if you need to adjust the waypoint you have the ref

Reply:
I can't say I'm a fan of that approach for the following reasons:
- Metadata can have additional attributes like description which would also have to be duplicated from the waypoint record
- If updating a waypoint name then you have to go and update all references as well.

A reference will point to the record with all the information.

Reply:
I think from a client's view defining response with `oneOf` is a bit problematic, as the client has no way to tell what it wants, and the spec allows the server to return whatever it wants. 

Having both `name` and `href` always present does not make it mandatory that other metadata attributes need to be included. We are just defining the minimum that clients can expect. Anything extra is ...extra.

Imho there's a clear use case for name, so that for the typical use case of displaying the route on map or as a list of named waypoints the client gets everything in one go.

I am ok with backtracking on this so that href is not included. If the client needs more details than waypoint names it can retrieve waypoints' data separately. I was just thinking that `name` is not guaranteed to be unique and as such does not work as an id. It would be sometimes handy to have the href as an id and as a reference that you can use to manipulate an individual waypoint.

So I am ok with
- both always
- just name
take your pick!

Reply:
The way I was seeing it is the metadata contains either:
1. A name (and possibly other attributes). Name is not an `id`... It's just a name to display.
2. An href which is a pointer to another  resource record (i.e. `waypoint`) which contains the relevant details.